app/Models/Printer.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use Illuminate\Support\Facades\Log;
+use Psr\Container\NotFoundExceptionInterface;
+use Psr\Container\ContainerExceptionInterface;
+use Symfony\Component\Process\Process;
+
+class Printer extends Model {
+    use HasFactory;
+
+    public $timestamps = false;
+
+    protected $fillable = [
+        'name',
+        'ip',
+        'port',
+        'paper_out_at',
+    ];
+
+    protected $casts = [
+        'paper_out_at' => 'datetime',
+    ];
+
+    /**
+     * Returns the `PrintJob`s that were executed by this printer.
+     * @return HasMany 
+     */
+    public function printJobs() {
+        return $this->hasMany(PrintJob::class);
+    }
+
+    /**
+     * Attemts to cancel the given `PrintJob`. Returns wether it was successful.
+     * @param PrintJob $printJob 
+     * @return PrinterCancelResult
+     */
+    public function cancelPrintJob(PrintJob $printJob) {
+        $process = new Process(['cancel', $printJob->job_id, '-h', "$this->ip:$this->port"]);
+        if (config('app.debug')) {
+            // cancel(1) exits with status code 0 if it succeeds
+            $result = ['output' => '', 'exit_code' => 0];
+        } else {
+            $process->run();
+            $result = ['output' => $process->getOutput(), 'exit_code' => $process->getExitCode()];
+        }
+        Log::info([$process->getCommandLine(), $result]);
+        if ($result['exit_code'] == 0) {
+            return PrinterCancelResult::Success;
+        }
+        if (strpos($result['output'], "already canceled") !== false) {
+            return PrinterCancelResult::AlreadyCancelled;
+        }
+        if (strpos($result['output'], "already completed") !== false) {
+            return PrinterCancelResult::AlreadyCompleted;
+        }
+        return PrinterCancelResult::CannotCancel;
+    }
+
+    /**
+     * Asks the printer to print a document with the given configuration.
+     * @param bool $twoSided 
+     * @param int $copies 
+     * @param string $path 
+     * @return int The `jobId` belonging to the printjob
+     * @throws PrinterException If the printing fails
+     */
+    public function print(bool $twoSided, int $copies, string $path) {
```

Comment:
I thought about doing it like that, but I think we would need to alter the db structure a bit. Because a PrintJob not only saves the printjob itself, but also its cost etc. So I think it would be a bit strange to put all of that cost calculation and print account modification into the Printer model aswell.